custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -138,6 +151,139 @@ def handle_exception(self, exception, repeat_record):
+                },
+            )
+        except NikshayResponseException as e:
+            _save_error_message(payload_doc.domain, payload_doc.case_id, e.message)
+
+    def handle_failure(self, response, payload_doc, repeat_record):
+        _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(response.json()))
+
+
+class NikshayHIVTestPayloadGenerator(BasePayloadGenerator):
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, person_case):
+        """
+        "PatientID": "Nikshay-ID",
+        "HIVStatus": "Pos/Neg/Unknown",
+        "HIVTestDate": "dd/mm/yyyy",
+        "CPTDeliverDate": "dd/mm/yyyy",
+        "ARTCentreDate": "dd/mm/yyyy",
+        "InitiatedOnART": 1/0, (Passing this 0 and not passing ARTCentreDate and InitiatedDate does not work)
+        "InitiatedDate": "dd/mm/yyyy",
+        "Source": "nikshay-id",
+        "regby": "nikshay-user",
+        "MORemark": "This is dev test", (Optional)
+        "IP_FROM": "Servers IP address or Mobile device MAC Address",
+        """
+
+        episode_case = get_open_episode_case_from_person(person_case.get_id)
+        episode_case_properties = episode_case.dynamic_case_properties()
+        person_case_properties = person_case.dynamic_case_properties()
+        properties_dict = {
+            "PatientID": episode_case_properties.get('nikshay_id'),
+            "HIVStatus": hiv_status.get(person_case_properties.get('hiv_status')),
+            "HIVTestDate": datetime.datetime.strptime(person_case_properties.get('hiv_test_date'),
+                                                      '%Y-%m-%d').strftime('%d/%m/%Y'),
+            # might not be available if cpt_initiated is no
+            "CPTDeliverDate": datetime.datetime.strptime(person_case_properties.get('cpt_initiation_date'),
+                                                         '%Y-%m-%d').strftime('%d/%m/%Y'),
+            # might not be available if art_initiated is no
+            "ARTCentreDate": datetime.datetime.strptime(person_case_properties.get('art_initiation_date'),
+                                                        '%Y-%m-%d').strftime('%d/%m/%Y'),
+            "InitiatedOnART": art_initiated.get(person_case_properties.get('art_initiated')),
+            # might not be available if art_initiated is no
+            "InitiatedDate": datetime.datetime.strptime(person_case_properties.get('art_initiation_date'),
+                                                        '%Y-%m-%d').strftime('%d/%m/%Y'),
+            "Source": ENIKSHAY_ID,
+            "regby": "tbu-dmdmo01",  # TODO: change this to a real username, store in localsettings
+            "IP_FROM": "127.0.0.1",
+        }
+
+        return json.dumps(properties_dict)
+
+    def handle_success(self, response, payload_doc, repeat_record):
+        # Simple success message that has {"Nikshay_Message": "Success"...}
+        pass
```

Comment:
Considering updating `hiv_test_nikshay_registered` here on the payload_doc which is a person unless there is specific case(may be, test) that is present for HIV test done.